lib/fluent/plugin/out_influxdb.rb

```diff
@@ -131,9 +135,30 @@ def write(chunk)
         :values    => values,
         :tags      => tags,
       }
-      points << point
+      retention_policy = nil
+      unless @retention_policy_key.nil?
+        retention_policy = record.delete(@retention_policy_key)
+        unless points.nil?
+          if !retention_policy.nil? && retention_policy != @default_retention_policy
+            # flush the retention policy first
+            @influxdb.write_points(points, nil, @default_retention_policy)
```

Comment:
In the above condition, `retention_policy` is not nil but set `@default_retention_policy` here?